transcom/mymove
# MB-2329 Revoke sessions via admin interface

One of the STIG requirements is the ability to revoke individual
sessions, and because we don't have access to the Redis server in
prod, we need to provide a way for someone to do this via the admin
website.

**Setup**:
`make db_dev_migrate`
`make redis_pull`

**To test session revocation:**

1. Sign in at http://milmovelocal:3000/ in one browser tab
2. In another browser tab, sign in at http://adminlocal:3000/ (You
might need to create yourself as an admin first via
http://adminlocal:3000/devlocal-auth/login)
3. Once in the admin site, click on Search User by ID on the left
sidebar
4. Enter your UUID and click the Search button. You can find your UUID
in the server logs in the middleware/request_logger.go:85 line under
the key `user-id`.
5. Once your details appear in the user show page, click on Edit in
the top right
6. Select "Yes" from the dropdown for "Revoke mil session"
7. Go back to the tab where you were signed into the mil app and
refresh it.
8.Verify that you are logged out, and that the session token has been
removed from the `current_mil_session_id` field in the Users table,
and from Redis (accessed via `redis-dev`, then list all keys with
`KEYS *`)

## Description

Explain a little about the changes at a high level.

## Reviewer Notes

Is there anything you would like reviewers to give additional scrutiny?

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
echo "Code goes here"
```

## Code Review Verification Steps

 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/backend#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:

## References

* [Jira story](tbd) for this change
* [this article](tbd) explains more about the approach used.

## Screenshots

If this PR makes visible UI changes, an image of the finished UI can help reviewers and casual
observers understand the context of the changes. A before image is optional and
can be included at the submitter's discretion.

Consider using an animated image to show an entire workflow instead of using multiple images. You may want to use GIPHY CAPTURE for this! 📸

_Please frame screenshots to show enough useful context but also highlight the affected regions._
